torchgeo/trainers/byol.py

```diff
@@ -161,7 +159,7 @@ def __init__(
         model: Module,
         projection_size: int = 256,
         hidden_size: int = 4096,
-        layer: Union[str, int] = -2,
```

Comment:
When we index `self.model.children()`, I believe this is a `nn.ModuleList`, so only integer indices are allowed. So far we're only using integer indices in our code. I don't know of a reason why we need to support string indices.

Reply:
I believe there was an option here where you could look up a layer by name that I removed for simplicity and forgot to update